public/jira/raw/rollback/rollback.ts

```diff
@@ -0,0 +1,311 @@
+    return { Parent: { key: change.fromString || "" } };
+  },
+  "Epic Link": function (lastReturnValue: unknown, change: Change) {
+    return { Parent: { key: change.fromString || "" } };
+  },
+  Status: function (lastReturnValue, change, fieldName, { statuses }) {
+    if (statuses.ids.has(change.from || "")) {
+      return { [fieldName]: statuses.ids.get(change.from || "") };
+    } else if (statuses.names.has(change.fromString || "")) {
+      return { [fieldName]: statuses.names.get(change.fromString || "") };
+    } else {
+      console.warn("Can't find status", change.from, change.fromString);
+      return { [fieldName]: { name: change.fromString || "" } };
+    }
+  },
+};
+
+const fieldAlias: { [key: string]: string } = {
+  duedate: "Due date",
+  status: "Status",
+  labels: "Labels",
+  issuetype: "Issue Type",
+  // "summary": "Summary" // we don't want to change summary
+  "Fix Version": "Fix versions",
+};
+
+function getSprintsMapsFromIssues(issues: JiraIssue[]): FieldLookup<Sprint> {
+  const ids = new Map<string, Sprint>();
+  const names = new Map<string, Sprint>();
+  for (const issue of issues) {
+    for (const sprint of issue.fields.Sprint || []) {
+      ids.set(sprint.id, sprint);
+      names.set(sprint.name, sprint);
+    }
+  }
+  return { ids, names };
+}
+
+function getVersionsFromIssues(issues: JiraIssue[]): FieldLookup<FixVersion> {
+  const ids = new Map<string, FixVersion>();
+  const names = new Map<string, FixVersion>();
+  for (const issue of issues) {
+    for (const version of issue.fields["Fix versions"]) {
+      ids.set(version.id, version);
+      names.set(version.name, version);
+    }
+  }
+  return { ids, names };
+}
+
+function getStatusesFromIssues(issues: JiraIssue[]): FieldLookup<Status> {
+  const ids = new Map<string, Status>();
+  const names = new Map<string, Status>();
+  for (const issue of issues) {
+    ids.set(issue.fields.Status.id, issue.fields.Status);
+    names.set(issue.fields.Status.name, issue.fields.Status);
+  }
+  return { ids, names };
+}
+
+export function rollbackIssues(
+  issues: JiraIssue[],
+  rollbackTime: Date = oneHourAgo
+): RolledBackJiraIssue[] {
+  const sprints = getSprintsMapsFromIssues(issues);
+  const versions = getVersionsFromIssues(issues);
+  const statuses = getStatusesFromIssues(issues);
+  return issues
+    .map((i) => rollbackIssue(i, { sprints, versions, statuses }, rollbackTime))
+    .filter((i): i is RolledBackJiraIssue => i !== undefined);
+}
+
+const oneHourAgo = new Date(Date.now() - 1000 * 60 * 60);
+
+/**
+ * @typedef {{
+ *   rolledvackTo: Date,
+ *   didNotExist: Boolean
+ * }} RolledBackMetadata
+ */
+
+/**
+ * @typedef {import("../../shared/types").JiraIssue & {rollbackMetadata: RolledBackMetadata}} RolledBackJiraIssue
+ */
+
+/**
+ * @param {import("../../shared/types").JiraIssue} issue
+ * @param {RollbackLookupData} data
+ * @param {Date} rollbackTime
+ * @returns {RolledBackJiraIssue | undefined}
+ */
```

Comment:
Do you think we still need the js doc comments?

Reply:
I did not want to remove it until we decide if we are going to replace them with ts docs or not
I just updated it based on the new types.

I do not like it anyways, I usually depend on the function signature to understand the parameters. So I am ok with removing it

Reply:
Removed